### PR TITLE
CB-18370, CB-19051 - Redbeams should return flow information

### PR DESCRIPTION
**Pull request by pdarvasi**
This commit:
- removes the calling of the unnecessary empty Redbeams RDS upgrade flow steps
  - TODO: the related Event, Request and Response classes are left there purposefully and to be deleted once we are sure that it won't cause any issues (meaning: in a subsequent release)
- eliminates the `resourceid` being 0 for a couple of flows in `flowlog` entity
- sets `DBStackService` as `ResourceIdProvider`
- introduces `UpgradeDatabaseResponse` that includes a `flowIdentifier`, `currentVersion`
and a `reason` if the flow is not triggerable due to a certain validation
-  when upgrade flow is already happening just returns the flowId and a descriptive reason field
- if the DB engine version matches the one sent in the request then it just returns the current version and a descriptive reason field

See detailed description in the commit message.

### CODE REVIEW

File: redbeams/src/main/java/com/sequenceiq/redbeams/controller/v4/progress/ProgressV4Controller.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 @Controller
 @Transactional(Transactional.TxType.NEVER)
-public class ProgressV4Controller implements ProgressV4Endpoint {
+    public class ProgressV4Controller implements ProgressV4Endpoint {
```
Comment (lacikaaa) [inline redbeams/src/main/java/com/sequenceiq/redbeams/controller/v4/progress/ProgressV4Controller.java:20]: could you remove the indentation?

File: redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/upgrade/RedbeamsUpgradeFlowConfig.java [hunk 2]
```diff
@@ -30,10 +26,8 @@ public class RedbeamsUpgradeFlowConfig extends AbstractFlowConfiguration<Redbeam
 
     private static final List<Transition<RedbeamsUpgradeState, RedbeamsUpgradeEvent>> TRANSITIONS =
             new Transition.Builder<RedbeamsUpgradeState, RedbeamsUpgradeEvent>().defaultFailureEvent(REDBEAMS_UPGRADE_FAILED_EVENT)
-                    .from(INIT_STATE).to(BACKUP_DATABASE_SERVER_STATE).event(REDBEAMS_START_UPGRADE_EVENT).defaultFailureEvent()
-                    .from(BACKUP_DATABASE_SERVER_STATE).to(UPGRADE_DATABASE_SERVER_STATE).event(BACKUP_DATABASE_SERVER_FINISHED_EVENT).defaultFailureEvent()
-                    .from(UPGRADE_DATABASE_SERVER_STATE).to(RESTORE_DATABASE_SERVER_STATE).event(UPGRADE_DATABASE_SERVER_FINISHED_EVENT).defaultFailureEvent()
-                    .from(RESTORE_DATABASE_SERVER_STATE).to(REDBEAMS_UPGRADE_FINISHED_STATE).event(RESTORE_DATABASE_SERVER_FINISHED_EVENT).defaultFailureEvent()
+                    .from(INIT_STATE).to(UPGRADE_DATABASE_SERVER_STATE).event(REDBEAMS_START_UPGRADE_EVENT).defaultFailureEvent()
+                    .from(UPGRADE_DATABASE_SERVER_STATE).to(REDBEAMS_UPGRADE_FINISHED_STATE).event(UPGRADE_DATABASE_SERVER_FINISHED_EVENT).defaultFailureEvent()
```
Comment (lacikaaa) [inline redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/upgrade/RedbeamsUpgradeFlowConfig.java:30]: this seems to be a non backward compatible change. what would happen to those upgrades which are already in these state during an upgrade?

File: redbeams/src/main/resources/schema/app/20220927124116_CB-18629_-_All_flow_logs_should_have_non_zero_resource_id.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
+-- // CB-18629 - All flow logs should have non zero resource id
+-- Migration SQL that makes the change goes here.
+UPDATE flowlog f
+SET resourceid =
+    (
+    SELECT MAX(resourceid)
+    FROM flowlog
+    WHERE f.flowid = flowid
+    )
+WHERE resourceid = 0;
```
Comment (lacikaaa) [inline redbeams/src/main/resources/schema/app/20220927124116_CB-18629_-_All_flow_logs_should_have_non_zero_resource_id.sql:10]: do we really need this for finalized flows?

Comment (pdarvasi) [reply redbeams/src/main/resources/schema/app/20220927124116_CB-18629_-_All_flow_logs_should_have_non_zero_resource_id.sql:10]: I have not tested filtering for non-finalized but finalized get purged periodically so it should not matter.

**Review by lacikaaa — APPROVED** [review]

**Review by daszabo — APPROVED** [review]

**Comment by pdarvasi** [comment]
retest this please

**Pull request merged**